### PR TITLE
Add spritz

### PR DESCRIPTION
**Pull request by acesnik**
Adding Spritz, a tool for generating proteogenomic databases from RNA-Seq to be used for searching proteomic data.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by acesnik** [comment]
@BiocondaBot update

**Comment by acesnik** [comment]
@BiocondaBot please add label

File: recipes/spritz/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  url: https://github.com/smith-chem-wisc/Spritz/releases/download/{{ version }}/SpritzCMD.zip
+  sha256: f218e3ebb599b29960ae0245aa4a16ab8734ed7895d98f0772b1f7734dacfc10
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - sysroot_linux-64 2.17  # [linux64]
```
Comment (bgruening) [inline recipes/spritz/meta.yaml:17]: why is this needed?

File: recipes/spritz/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  run:
+    - snakemake-minimal =5.27.4
+    - mamba
```
Comment (bgruening) [inline recipes/spritz/meta.yaml:20]: mamba and conda can be removed afaik, or are you installing dependencies during tool runtime?

File: recipes/spritz/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    - python=3.8
```
Comment (bgruening) [inline recipes/spritz/meta.yaml:21]: any reason why you pin python and dotnet?

File: recipes/spritz/meta.yaml (hunk 2 above)
Comment (acesnik) [reply recipes/spritz/meta.yaml:20]: Yes, Spritz installs dependencies during tool runtime using snakemake with conda/mamba.

File: recipes/spritz/meta.yaml (hunk 3 above)
Comment (acesnik) [reply recipes/spritz/meta.yaml:21]: Spritz's commandline tool uses dotnet 5.0, and it doesn't work with other frameworks/runtimes, so that needs to be pinned.

Comment (acesnik) [reply recipes/spritz/meta.yaml:21]: It needs python 3. Should I just remove the pin, or pin it to `python=3`?

File: recipes/spritz/meta.yaml (hunk 2 above)
Comment (bgruening) [reply recipes/spritz/meta.yaml:20]: Why does needs to happen during tool runtime? Please be aware that cluster installations exists, where conda is read-only from worker nodes.

File: recipes/spritz/meta.yaml (hunk 1 above)
Comment (acesnik) [reply recipes/spritz/meta.yaml:17]: These are prerequisites accompanying dotnet's need for CentOS 7.

See:
https://conda-forge.org/docs/maintainer/knowledge_base.html#using-centos-7
https://github.com/conda-forge/dotnet-feedstock/blob/master/conda-forge.yml#L5
https://github.com/conda-forge/dotnet-feedstock/blob/master/recipe/meta.yaml#L47

Do you think I should remove this, since CentOS 7 is the default container for bioconda now?

File: recipes/spritz/meta.yaml (hunk 3 above)
Comment (bgruening) [reply recipes/spritz/meta.yaml:21]: We do not build for python2 anymore. So you can really remove that.

File: recipes/spritz/meta.yaml (hunk 2 above)
Comment (acesnik) [reply recipes/spritz/meta.yaml:20]: Spritz has three workflows right now, and not all tools are required if users just need one of the three. In my tests, this is more efficient, not least because spritz uses mamba and users tend to use conda installs by default. Spritz has an option for protected-access cluster setups, where resources (e.g. uniprot, ensembl databases) are downloaded beforehand. I believe snakemake, which is part of the backend of spritz, sets up environments (e.g. using pre-downloaded bioconda/conda-forge repositories) and then activates them before firing up worker nodes. I've tested this via `slurm` on a cluster available to me. Are there other cluster setups that are problematic here?

File: recipes/spritz/meta.yaml (hunk 1 above)
Comment (acesnik) [reply recipes/spritz/meta.yaml]: I'll give this recipe a try with these removed.

File: recipes/spritz/meta.yaml (hunk 3 above)
Comment (acesnik) [reply recipes/spritz/meta.yaml]: I'm giving it a try with these pins removed, other than for `dotnet-runtime`.

File: recipes/spritz/meta.yaml (hunk 2 above)
Comment (acesnik) [reply recipes/spritz/meta.yaml:17]: Thanks for the heads-up on the possible cluster installation issues, by the way. I'm happy to do some more testing of that issue before merging if that would be best and if I can find such an environment.

Comment (bgruening) [reply recipes/spritz/meta.yaml:17]: All I can say is that installation of packages during runtime is not usual, a dependency on conda/mamba as runtime dependency is not usual. Why not splitting up your workflows into 3 packages? We don't have strict rules here about that I just wanted to point it out and this might cause problems down the road for your users.

Comment (acesnik) [reply recipes/spritz/meta.yaml:17]: Makes sense!

The workflows aren't separate; they can be combined. For example, users can make databases with variants or transcript isoforms, or they can request both variants and isoforms together which yields a different result than they do separately. Similarly, users can request to do quantification of these features or not.

Comment (bgruening) [reply recipes/spritz/meta.yaml:19]: But Bioconda is not a workflow repository but a tool repository :)
I know the lines a blurry, but this is the reason repos and communities like snakepipes, nf-core etc exists. Workflows are a different beast.

Comment (acesnik) [reply recipes/spritz/meta.yaml:19]: Oh that's an interesting point. Snakepipes does look like it might be a more appropriate home for Spritz.

**Comment by acesnik** [comment]
I'm going to close this for now and see about integrating spritz into snakepipes.
https://github.com/bioconda/bioconda-recipes/pull/29875#discussion_r685093297
https://github.com/maxplanck-ie/snakepipes

**Comment by acesnik** [comment]
Hi @bgruening, just as an update, it sounds like snakepipes doesn't consider Spritz something they would consider merging. They don't view it as a general repository for snakemake workflows. https://github.com/maxplanck-ie/snakepipes/issues/782
